hack/generator/pkg/codegen/pipeline_augment_status_test.go

```diff
@@ -0,0 +1,36 @@
+package codegen
+
+import (
+	. "github.com/onsi/gomega"
+
+	"testing"
+)
+
+func Test_ShouldSkipDir_GivenPath_HasExpectedResult(t *testing.T) {
+	cases := []struct {
+		name       string
+		path       string
+		shouldSkip bool
+	}{
+		// Simple paths
+		{"Root", "/", false},
+		{"Top level", "/foo/", false},
+		{"Nested", "/foo/bar/", false},
+		// Paths to skip
+		{"Skip top level", "/examples/", true},
+		{"Skip nested", "/foo/examples/", true},
+		{"Skip nested, trailing directory", "/foo/examples/bar/", true},
+	}
```

Comment:
Could you add some test cases for Windows-y paths? Or would those fail on non-Windows machines?

Reply:
Good idea, added.

Reply:
I've added them - but they fail when run on Linux, so I've gated them to only run on Windows. #sigh